### DISCUSSION TITLE
Declaring an `element` in the target url for a link

### DISCUSSION THREAD

**Discussion by bozdevs** [swup/swup#909] [General]
This is a very important situation, but I couldn't figure out how to do it. What I want to do is exactly this:

- Imagine there are multiple products listed on the `/products/` page.
- You can imagine that the product detail pages are `/products/{product_name}`.
- When we click on the link of one of the products listed on the `/products/` page, **swup** goes to the relevant detail page.
- However, I want to show the information there via a _(modal, pop-up)_, without going to the product's detail page. For example, displaying comments about the product.

In this situation;

```html
<!-- here is "/products/" page -->
<a 
   href="/products/t-shirt/" 
   data-source="#comments" 
   data-target="#modal" 
   onclick="...remove_display_none_from_modal..."
>
   See Comments
</a>
<div id="modal" style="display: none">
   Loading...
</div>
```

----

### To explain this situation more simply;

Have a login page at `"/login/"`. Contain a login form `<form>...</form>`.

In some cases, when the "login" button is clicked, instead of going to "/login/" and changing the page, you stay on the same page, target the `<form>...</form>` element on the target page, and open it in a `"modal"` within the page you are currently on.

----

I tried to create a structure like above, but I could not achieve this with the **"Fragment Plugin"**. I also tried various scenarios by listening to the `"swup:visit:start"` event, but I failed.

Is it possible to do this? Please someone help me.

**Comment by daun** [comment]
That's where the Fragment Plugin really shines. There's a [simple demo app showing modal support](https://swup.js.org/getting-started/demos/#fragment-support-modal) in the docs, and there's also a much more [comprehensive demo showing modal and filter support](https://swup-fragment-plugin.netlify.app/) from the plugin itself.

If you've tried implementing from those examples, feel free to share what you've got so far so others can chime in.
